scripts/sudo.ps1

```diff
@@ -1,87 +1,94 @@
+<#
+	.SYNOPSIS
+		Runs a scriptblock, command or application as an elevated process using sudo.exe
+	.DESCRIPTION
+		Wraps sudo.exe to add functionality for running PowerShell scripts or commands in an elevated process.
+
+		When running a scriptblock or PowerShell command, a new copy of PowerShell is run with sudo and an EncodedCommand.
+		When running a native application, the command itself is run with sudo.
+
+		This script DOES NOT (currently) support piping input to the elevated command.
+#>
+
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
+[CmdletBinding(DefaultParameterSetName = "Script")]
+param(
+	# A scriptblock to run in an elevated process
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Script")]
+	[scriptblock]$ScriptBlock,
 
-# open question - Should -NoProfile be used when invoking PowerShell
-BEGIN {
-	if ($__SUDO_TEST -ne $true) {
-		$SUDOEXE = "sudo.exe"
-	}
-	else {
-		if ($null -eq $SUDOEXE) {
-			throw "variable SUDOEXE has not been set for testing"
-		}
-	}
+	# Run PowerShell with the -NoProfile switch
+	[switch]$NoProfile,
 
-	if ([Environment]::OSVersion.Platform -ne "Win32NT") {
+	# A command or application to run in an elevated process
+	[Parameter(Mandatory, Position = 0, ParameterSetName = "Command")]
+	[string]$Command,
+
+	# Arguments to pass to the command or application
+	[Parameter(Position = 1, ParameterSetName = "Command", ValueFromRemainingArguments)]
+	[Alias("Args")]
+	[PSObject[]]$ArgumentList
+)
+begin {
+	if ($IsLinux -or $IsMacOS) {
 		throw "This script works only on Microsoft Windows"
 	}
 
-	if ($null -eq (Get-Command -Type Application -Name "$SUDOEXE" -ErrorAction Ignore)) {
-		throw "'$SUDOEXE' cannot be found."
+	if ($__SUDO_TEST -ne $true) {
+		$Env:SUDOEXE = "sudo.exe"
```

Comment:
This seems overly complicated (you have to set _two_ variables (SUDOEXE and __SUDO_TEST) to actually be able to use an alternate sudo.exe for testing? I'm not sure why this was here.

I would prefer to just have a _parameter_ `SudoPath` that you can set to whatever you want for testing purposes, but that defaults to just `sudo.exe` and assumes that sudo.exe is in the PATH. 

NOTE: I changed SUDOEXE to an environment variable. I'd rather remove it. If we want to keep it, should SUDO_TEST also be an environment variable?